imap_processing/tests/spice/test_geometry.py

```diff
@@ -306,3 +308,49 @@ def test_instrument_pointing(furnish_kernels):
             et, SpiceFrame.IMAP_HI_90, SpiceFrame.ECLIPJ2000, cartesian=True
         )
         assert ins_pointing.shape == (3, 3)
+
+
+def test_cartesian_to_spherical():
+    """Tests cartesian_to_spherical function."""
+    v = np.array(
+        [
+            [-186.5575, -707.5707, 618.0569],
+            [508.5697, -516.0282, 892.6931],
+            [508.5697, -516.0282, 892.6931],
+            [508.5697, -516.0282, 892.6931],
+        ]
+    )
+
+    spherical_coords = cartesian_to_spherical(v)
+    az_sc, el_sc, _ = (
+        spherical_coords[..., 0],
+        spherical_coords[..., 1],
+        spherical_coords[..., 2],
+    )
+
+    np.testing.assert_allclose(
+        np.unique(np.radians(az_sc)), np.array([1.31300, 2.34891]), atol=1e-05, rtol=0
+    )
+    np.testing.assert_allclose(
+        np.unique(np.radians(el_sc)), np.array([-0.88901, -0.70136]), atol=1e-05, rtol=0
+    )
+
+
+def test_spherical_to_cartesian():
```

Comment:
I would love to see this test bulked up by using a meshgrid of `theta, phi` values in the range `[0, pi]` and `[-pi, pi]` with steps of something like `pi/50` so that all directions are covered. Results can be checked against the spiceypy function `sphrec`.